From 4980d629d906786cdfc5bd2ed4e61b4f8676b6bd Mon Sep 17 00:00:00 2001
From: Chris Feist <cfeist@redhat.com>
Date: Tue, 5 Aug 2014 16:17:45 -0500
Subject: [PATCH 02/10] Ignore SIGPIPE so piping output of ccs won't cause an
 IOError and traceback

---
 ricci/ccs/ccs | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ricci/ccs/ccs b/ricci/ccs/ccs
index 8ea37a9..95e011c 100755
--- a/ricci/ccs/ccs
+++ b/ricci/ccs/ccs
@@ -13,6 +13,7 @@ from Queue import Queue
 import getpass
 import subprocess
 import tempfile
+from signal import signal, SIGPIPE, SIG_DFL
 
 RICCI_PORT = 11111
 CLUSTERRNG = "/usr/share/ccs/cluster.rng"
@@ -101,6 +102,7 @@ def main(argv):
     global verifyconf, schema, nounfence
     global noenable
 
+    signal(SIGPIPE,SIG_DFL)
     noenable = False
 #    logging.basicConfig(level=logging.DEBUGRemove
     try:
-- 
2.1.0

