diff --git a/ricci/ccs/ccs b/ricci/ccs/ccs
index 4f00e11..bcc833c 100755
--- a/ricci/ccs/ccs
+++ b/ricci/ccs/ccs
@@ -1502,12 +1502,19 @@ def list_serviceopts(options):
     else:
         servicename = options[0]
 
+    # We don't print out info about the 'action' service
+    if servicename == "action":
+        return
+
     rng = open(CLUSTERRNG)
     dom = minidom.parseString(rng.read())
 
     for elem in dom.getElementsByTagName("define"):
         if elem.getAttribute("name") == "CHILDREN":
             for elem2 in elem.getElementsByTagName("ref"):
+                # We don't print out info about the 'action' service
+                if elem2.getAttribute("name") == "RESOURCEACTION":
+                    continue
                 if servicename:
                     if elem2.getAttribute("name").upper() == servicename.upper():
                         (servname, servopts) = getServiceOptions(dom, elem2)
@@ -2058,13 +2065,7 @@ def send_to_ricci(msg, host=hostname):
     cert = os.path.expanduser("~/.ccs/cacert.pem")
     key = os.path.expanduser("~/.ccs/privkey.pem")
     config = os.path.expanduser("~/.ccs/cacert.config")
-    try:
-        (family, socktype, proto, canonname, sockaddr) = socket.getaddrinfo(host, RICCI_PORT)[0]
-    except socket.gaierror:
-        print "Unable to resolve %s, does that host exist?" % host
-        sys.exit(1)
-    
-    s = socket.socket(family, socktype, proto)
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
 
     # Make sure we have a client certificate and private key
     # If not we need to autogenerate them (including creating an
